extensions/wikibase/src/org/openrefine/wikibase/editing/MediaFileUtils.java

```diff
@@ -77,8 +78,31 @@ public void purgePage(long pageid) throws IOException, MediaWikiApiErrorExceptio
         Map<String, String> parameters = new HashMap<>();
         parameters.put("action", "purge");
         parameters.put("pageids", Long.toString(pageid));
-
-        apiConnection.sendJsonRequest("POST", parameters);
+        int retries = 5;
+        int backofTime = maxLagWaitTime;
+        while (retries > 0) {
+            try {
+                JsonNode response = apiConnection.sendJsonRequest("POST", parameters);
+                if (!(response != null && response.has("warnings"))) {
+                    return;
```

Comment:
Not every warning is necessarily a rate-limiting one. Is there a warning code we could rely on to check if it's a rate-limiting one? There might be warnings that are not worth retrying (such as insufficient editing permissions, lack of authentication…).

Reply:
There is no code; i did not include the check for the rate limit as I was not sure if the warning text is internationalised. Any guidance on this?

Reply:
I agree, if it's just text it feels too brittle to rely on it indeed.

Reply:
According to https://www.mediawiki.org/wiki/API:Etiquette#Request_limit the response should contain an error (not a warning) with the code `ratelimited`.

Reply:
@sebastian-berlin-wmse  As mentioned in the link  API returns an error only when it is a edit call --> _Requests which make edits, modify state or otherwise are not read-only requests, are subject to rate limiting._

In case of Delete / Purge call the rate limit is returned as a warning.

Reply:
Looks like it. I didn't think `purge` was counted as read-only, but apparently it does.

You can still get a code for the warning if you add parameter `errorformat=raw`. That'll give you a response like:
```
{
  "batchcomplete": "",
  "warnings": [
    {
      "code": "ratelimited",
      "key": "apierror-ratelimited",
      "params": [],
      "module": "purge"
    }
  ],
  "purge": [
    {
      "ns": 0,
      "title": "Main Page"
    }
  ]
}
```

Reply:
Updated code to pass param errorformat and validating code in response to identify if it is a rate limit error.